### ISSUE TITLE
Recognize that Assert.ThrowsAsync etc now return a Task that needs awaiting.

### ISSUE THREAD

**Issue by manfred-brands** [nunit/nunit.analyzers#988]
In NUnit 5, the `Async` named method are actually converted to `async` and now return a `Task`

This rule is to ensure that users `await` the task instead of ignoring it meaning the code isn't actually executed.

```csharp
        [Test]
        public void GenericThrowsAsyncSucceedsWithDelegate()
        {
            Assert.ThrowsAsync<ArgumentException>(AsyncTestDelegates.ThrowsArgumentException);
        }
```

The analyzer should detect this and offer a code fix to not only `await` the method but also potentially change the return type to `async Task`:

```csharp
        [Test]
        public async Task GenericThrowsAsyncSucceedsWithDelegate()
        {
            await Assert.ThrowsAsync<ArgumentException>(AsyncTestDelegates.ThrowsArgumentException);
        }
```
